### ISSUE TITLE
A tensorflow version of DirectAU

### ISSUE THREAD

**Issue by ll0ruc** [THUwangcy/DirectAU#8]
Hi, a very attractive work, have you tried changing  DirectAU to tensorflow version? 
I tried this but it takes a long time [almost 5min per epoch].

**Comment by THUwangcy** [comment]
Haven't yet. The main bottleneck maybe the calculation of the uniformity loss. A large batch size will make the training process slow. So make sure the batch size is not very large. Generally 256~1024 is fine.

Besides, I'm not sure whether tensorflow has a similar function as `torch.pdist`. Directly using loops to calculate uniformity is also inefficient.

**Comment by ll0ruc** [comment]
Thank you,  I have adjusted a fast version
